hooks/VDM.js

```diff
@@ -268,19 +304,304 @@ module.exports = function (input, session) {
+        path: `environment.observations.${meteoLocation}.water.levelTrend`,
+        value: statusTable[data.waterlevelten],
+      })
+    }
+  }
+
+  if (data.surfcurrspd) {
+    contextPrefix = 'meteo.'
+    if (data.surfcurrspd < 252) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.surfaceCurrent.speed`,
+        value: utils.transform((data.surfcurrspd / 10), 'knots', 'ms'),
+      })
+    }
+  }
+
+  if (data.surfcurrdir) {
+    contextPrefix = 'meteo.'
+    if (data.surfcurrdir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.surfaceCurrent.direction`,
+        value: utils.transform(data.surfcurrdir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.signwavewhgt) {
+    contextPrefix = 'meteo.'
+    if (data.signwavewhgt < 252) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wave.significantHeight`,
+        value: data.signwavewhgt / 10,
+      })
+    }
+  }
+
+  if (data.waveperiod) {
+    contextPrefix = 'meteo.'
+    if (data.waveperiod < 61) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wave.period`,
+        value: data.waveperiod,
+      })
+    }
+  }
+
+  if (data.wavedir) {
+    contextPrefix = 'meteo.'
+    if (data.wavedir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wave.direction`,
+        value: utils.transform(data.wavedir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.swellhgt) {
+    contextPrefix = 'meteo.'
+    if (data.swellhgt < 252) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.height`,
+        value: data.swellhgt / 10,
+      })
+    }
+  }
+
+  if (data.swellperiod) {
+    contextPrefix = 'meteo.'
+    if (data.swellperiod < 61) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.period`,
+        value: data.swellperiod,
+      })
+    }
+  }
+
+  if (data.swelldir) {
+    contextPrefix = 'meteo.'
+    if (data.swelldir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.direction`,
+        value: utils.transform(data.swelldir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.seastate) {
+    contextPrefix = 'meteo.'
+    if (data.seastate < 13) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.seaState`,
+        value: beaufortScale[data.seastate],
+      })
+    }
+  }
+
+  if (data.watertemp) {
+    contextPrefix = 'meteo.'
+    if (data.watertemp > -101 && data.watertemp < 501) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.water.temperature`,
+        value: utils.transform((data.watertemp / 10), 'c', 'k'),
+      })
+    }
+  }
+
+  if (data.precipitation) {
+    contextPrefix = 'meteo.'
+    if (data.precipitation < 7) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.precipitation`,
```

Comment:
Does the type of precipitation need to be in the path e.g. `environment.observations.precipitation.rain`,  `environment.observations.precipitation.snow`?
Just a thought as weather services seem to differentiate and might be a consideration when incorporating forecasat data.

Reply:
Lookup table for value are here. Just another weather data.
https://github.com/KEGustafsson/nmea0183-signalk/blob/met-hyd-data/hooks/VDM.js#L98-L107

Reply:
In this case, should the path then be `precipitationType` then if a forecast has a `volume` of precepitation it could be represented like this: 
```
{
   ...
          {
            "path": "environment.air.precipitationType",
            "value": 7
          },
          {
            "path": "environment.air.precipitationVolume",
            "value": 12
          },
   ...
}
```

Reply:
There is just one topic for precipitation and now it is linked to air. Nothing else.